test/test-debuglet.ts

```diff
@@ -871,6 +871,34 @@ describe('Debuglet', function() {
       debuglet.start();
     });
 
+    it('should resolve promises before executing user functions', function(done) {
+      this.timeout(2000);
+      const debug = new Debug(
+          {projectId: 'fake-project', credentials: fakeCredentials});
+      const debuglet = new Debuglet(debug, defaultConfig);
+
+      const scope = nock(API)
+                      .post(REGISTER_PATH)
+                      .reply(200, {debuggee: {id: DEBUGGEE_ID}})
+                      .get(BPS_PATH + '?successOnTimeout=true')
+                      .reply(200, {breakpoints: []});
+      debuglet.start();
+      const debugPromise = debuglet.checkReady();
+
+      debuglet.once('registered', function reg(id: string) {
+        assert.equal(id, DEBUGGEE_ID);
+        debugPromise.then(() => {
+          setTimeout(function() {
```

Comment:
Is `setTimeout` necessary here?

Reply:
I think so. This is a format to make sure nock.done is called to check that all specified calls on that scope were performed.